src/metabase/lib/card.cljc

```diff
@@ -162,5 +162,5 @@
                         (assoc metric-query :lib/metadata (:lib/metadata query))
                         -1
                         (lib.util/query-stage metric-query -1)
-                        options)))
+                        (dissoc options :unique-name-fn))))
```

Comment:
If unique-name-fn gets triggered in here, then the rest of returned columns will be thrown off.
I was wondering if this should still use `card-metadata-columns` with `dataset-query` and `results-metadata` replaced.  I think that affects whether named or id references are used.

Reply:
How will the rest of the returned columns be thrown off?

Reply:
Unsure, I was getting a `PRODUCT_ID_2` in the second join condition in this query. But I can't reproduce it now

<img width="1286" alt="image" src="https://github.com/metabase/metabase/assets/105012/2b89cb91-53b3-45ed-9a8c-f3373326e21a">